### PR TITLE
fixup changes from PR #407

### PR DESCRIPTION
**Pull request by mattip**
After the last commit to #407, there were some test failures. 
- When I rerun `make autogen`, it removes the implementation of `ctx_ContextVar_New` and `ctx_ContextVar_Set`, which then breaks the build. If I restore the mappings in `hpy/tools/autogen/conf.py`, then the build succeeds.

- Also there was a problem with doc building.

### CODE REVIEW

**Comment by mattip** [comment]
@fangerer

**Review by hodgestar — APPROVED** [review]

**Review by fangerer — APPROVED** [review]
Sorry, I thought the tests were green except of the well-known failure in `cppcheck`.

**Comment by mattip** [comment]
I am confused why I needed to add the mappings to `hpy/tools/autogen/conf.py`. I thought as long as the function name is `H` + <cpython name>`, then the `ctx_<name>` functions would be generated automatically. What did I miss?

File: hpy/tools/autogen/conf.py [hunk 1]
```diff
@@ -139,7 +139,7 @@
     'HPyBool_FromBool': 'PyBool_FromLong',
     'HPy_Compile_s': None,
     'HPy_EvalCode': 'PyEval_EvalCode',
-    'HPyContextVar_New': None,
+    'HPyContextVar_New': 'PyContextVar_New',
```
Comment (fangerer) [inline hpy/tools/autogen/conf.py:142]: Ah, I didn't properly parse the previous diff. My suggestion was to just remove lines 142 and 144.

**Review by fangerer — CHANGES_REQUESTED** [review]
Docs build is still failing. Line 70 of `docs/index.rst` must be:
```
    contributing/index
```

**Review by fangerer — APPROVED** [review]

**Comment by mattip** [comment]
Only cppcheck is failing. Here is the rendered doc page (without formatting), does that read OK?

<h1>Getting Started<a class="headerlink" href="[#getting-started](view-source:file:///home/matti/.cache/.fr-4ccs5Z/contributing/index.html#getting-started)" title="Permalink to this headline">¶</a></h1>
<p>TBD</p>
</div>
<div class="section" id="adding-new-api">
<h1>Adding New API<a class="headerlink" href="[#adding-new-api](view-source:file:///home/matti/.cache/.fr-4ccs5Z/contributing/index.html#adding-new-api)" title="Permalink to this headline">¶</a></h1>
<ol class="arabic simple">
<li><p>Add the function to <code class="docutils literal notranslate"><span class="pre">hpy/tools/autogen/public_api.h</span></code>. If the CPython
equivalent function name is not the same (after removing the leading <code class="docutils literal notranslate"><span class="pre">H</span></code>,
add an appropriate CPython function mapping in <code class="docutils literal notranslate"><span class="pre">hpy/tools/autogen/conf.py</span></code>.
If the signature is complicated or there is no clear equivalent function,
the mapping should be <code class="docutils literal notranslate"><span class="pre">None</span></code>, and follow the directions in the next step.
Otherwise all the needed functions will be autogenerated.</p></li>
<li><p>If the function cannot be autogenerated (i.e. the mapping does not exist),
you must write the wrapper by hand. Add the function to <code class="docutils literal notranslate"><span class="pre">NO_WRAPPER</span></code> in
<code class="docutils literal notranslate"><span class="pre">hpy/tools/autogen/debug.py</span></code>, and add a <code class="docutils literal notranslate"><span class="pre">ctx_fname</span></code> function to
<code class="docutils literal notranslate"><span class="pre">hyp/devel/src/runtime/*.c</span></code> (possibly adding the new file to <code class="docutils literal notranslate"><span class="pre">setup.py</span></code>),
add a debug wrapper to <code class="docutils literal notranslate"><span class="pre">hpy/debug/src/debug_ctx.c</span></code>, add a implementation
that uses the <cite>ctx</cite> variant to <code class="docutils literal notranslate"><span class="pre">hpy/devel/include/hpy/cpython/misc.h</span></code> and
add the declaration to <code class="docutils literal notranslate"><span class="pre">hpy/devel/include/hpy/runtime/ctx_funcs.h</span></code>.</p></li>
<li><p>Run <code class="docutils literal notranslate"><span class="pre">make</span> <span class="pre">autogen</span></code> which will turn the mapping into autogenerated functions</p></li>
<li><p>Add a test for the functionality</p></li>
<li><p>Build with <code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">setup.py</span> <span class="pre">build_ext</span></code>. After that works, build with
<code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">-m</span> <span class="pre">pip</span> <span class="pre">install</span> <span class="pre">-e</span> <span class="pre">.</span></code>, then run the test with <code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">-m</span> <span class="pre">pytest</span> <span class="pre">...</span></code>.</p>

**Pull request merged**